src/compiler/checker.ts

```diff
@@ -13540,6 +13540,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         result.compositeSignatures = concatenate(left.compositeKind !== TypeFlags.Intersection && left.compositeSignatures || [left], [right]);
         if (paramMapper) {
             result.mapper = left.compositeKind !== TypeFlags.Intersection && left.mapper && left.compositeSignatures ? combineTypeMappers(left.mapper, paramMapper) : paramMapper;
+        } 
+        else if (left.compositeKind !== TypeFlags.Intersection && left.mapper && left.compositeSignatures) {

```

Comment:
This whole function is very similar to `combineSignaturesOfIntersectionMembers` so I thought that there might be a similar problem there but so far I couldn't create a test case that would manifest an issue there.

It's also worth noting that both of those functions check `left.compositeKind` but it's never the "other one" in the existing test suite (in `combineSignaturesOfIntersectionMembers` we never witness `left.compositeKind === TypeFlags.Union` and in `combineSignaturesOfUnionMembers` we never witness `left.compositeKind === TypeFlags.Intersection`). So either some tests are missing or those branches are unused.